tests/integration/test_storage_nats/test_nats_jet_stream.py

```diff
@@ -0,0 +1,1766 @@
+
+
+def test_nats_insert(nats_cluster):
+
+    asyncio.run(add_durable_consumer(cluster, "test_stream", "test_consumer"))
+
+    instance.query(
+        f"""
+        CREATE TABLE test.nats (key UInt64, value UInt64)
+            ENGINE = NATS
+            SETTINGS nats_url = 'nats1:4444',
+                     nats_stream = 'test_stream',
+                     nats_subjects = 'test_subject',
+                     nats_format = 'TSV',
+                     nats_row_delimiter = '\\n';
+        """
+    )
+
+    values = []
+    for i in range(50):
+        values.append("({i}, {i})".format(i=i))
+    values = ",".join(values)
+
+    while True:
+        try:
+            instance.query("INSERT INTO test.nats VALUES {}".format(values))
+            break
+        except QueryRuntimeException as e:
+            if "Local: Timed out." in str(e):
+                continue
+            else:
+                raise
+    insert_messages = asyncio.run(receive_messages(nats_cluster, "test_stream", "test_consumer", "test_subject"))
+
+    result = "\n".join(insert_messages)
+    nats_helpers.check_result(result, True)
+
+
+def test_fetching_messages_without_mv(nats_cluster):
+
+    asyncio.run(add_durable_consumer(cluster, "test_stream", "test_consumer"))
+
+    instance.query(
+        f"""
+        CREATE TABLE test.nats (key UInt64, value UInt64)
+            ENGINE = NATS
+            SETTINGS nats_url = 'nats1:4444',
+                     nats_stream = 'test_stream',
+                     nats_subjects = 'test_subject',
+                     nats_format = 'TSV',
+                     nats_row_delimiter = '\\n';
+        """
+    )
+
+    values = []
+    for i in range(50):
+        values.append("({i}, {i})".format(i=i))
+    values = ",".join(values)
+
+    while True:
+        try:
+            instance.query("INSERT INTO test.nats VALUES {}".format(values))
+            break
+        except QueryRuntimeException as e:
+            if "Local: Timed out." in str(e):
+                continue
+            else:
+                raise
+    
+    insert_messages = asyncio.run(receive_messages(nats_cluster, "test_stream", "test_consumer", "test_subject"))
+    result = "\n".join(insert_messages)
+    nats_helpers.check_result(result, True)
+
+def test_nats_many_subjects_insert_wrong(nats_cluster):
+
+    instance.query(
+        f"""
+        CREATE TABLE test.nats (key UInt64, value UInt64)
+            ENGINE = NATS
+            SETTINGS nats_url = 'nats1:4444',
+                     nats_stream = 'test_stream',
+                     nats_subjects = 'insert1,insert2.>,insert3.*.foo',
+                     nats_format = 'TSV',
+                     nats_row_delimiter = '\\n';
+        """
+    )
+
+    values = []
+    for i in range(50):
+        values.append("({i}, {i})".format(i=i))
+    values = ",".join(values)
+
+    # no subject specified
+    instance.query_and_get_error("INSERT INTO test.nats VALUES {}".format(values))
```

Comment:
When using `query_and_get_error`, could you please do some checks on the returned error that it contains the error that you expect?